tools/test.py

```diff
@@ -114,16 +115,30 @@ def main():
 
     # build the model and load checkpoint
     model = build_detector(cfg.model, train_cfg=None, test_cfg=cfg.test_cfg)
-    fp16_cfg = cfg.get('fp16', None)
-    if fp16_cfg is not None:
-        wrap_fp16_model(model)
-    checkpoint = load_checkpoint(model, args.checkpoint, map_location='cpu')
-    if args.fuse_conv_bn:
-        from tools.fuse_conv_bn import fuse_module
-        model = fuse_module(model)
+
+    # nncf model wrapper
+    if cfg.get('nncf_config'):
+        check_nncf_is_enabled()
+        if not is_checkpoint_nncf(args.checkpoint):
+            raise RuntimeError('Trying to make testing with NNCF compression a model snapshot that was trained with NNCF')
+        cfg.load_from = args.checkpoint
+        cfg.resume_from = None
+        if torch.cuda.is_available():
+            model = model.cuda()
+        _, model = wrap_nncf_model(model, cfg, None, get_fake_input)
+        checkpoint = torch.load(args.checkpoint, map_location=None)
+    else:
+        fp16_cfg = cfg.get('fp16', None)
+        if fp16_cfg is not None:
+            wrap_fp16_model(model)
+        checkpoint = load_checkpoint(model, args.checkpoint, map_location='cpu')
+        if args.fuse_conv_bn:  # TODO: FIXME: should it be inside this 'else' branch???
+            from tools.fuse_conv_bn import fuse_module
+            model = fuse_module(model)
```

Comment:
2 @Ilya-Krylov:  
Ilya, could you give us your recommendation, if this `fuse_conv` operation should be inside this `else` branch (e.g. when  NNCF is not enabled)?    
I'm not sure, since I did not investigated thoroughly how it works indeed.

Reply:
To my knowledge we never used it. But from integration perspective I think NNCF should cover this case as well.
@druzhkov-paul what do you think?